components/apps_form/apps_form_field/apps_form_field.tsx

```diff
@@ -59,15 +59,7 @@ export default class AppsFormField extends React.PureComponent<Props> {
     handleSelected = (selected: AppSelectOption | UserProfile | Channel) => {
         const {name, field, onChange} = this.props;
 
-        if (field.type === AppFieldTypes.USER) {
-            const user = selected as UserProfile;
-            let selectedLabel = user.username;
-            if (this.props.teammateNameDisplay) {
-                selectedLabel = displayUsername(user, this.props.teammateNameDisplay);
-            }
-            const option = {label: selectedLabel, value: user.id};
-            onChange(name, option);
-        } else if (field.type === AppFieldTypes.CHANNEL) {
+        if (field.type === AppFieldTypes.CHANNEL) {
```

Comment:
As noted in https://github.com/mattermost/mattermost-server/issues/21484, this feature also needs to support multi-selecting channels